include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+                      std::string const &options = "IterInverse=0")
+            : Mapping(reinterpret_cast<AstMapping *>(
+                      _makeRawChebyMap(coeff_f, nout, lbnd_f, ubnd_f, options))) {}
+
+    virtual ~ChebyMap() {}
+
+    ChebyMap(ChebyMap const &) = delete;
+    ChebyMap(ChebyMap &&) = default;
+    ChebyMap &operator=(ChebyMap const &) = delete;
+    ChebyMap &operator=(ChebyMap &&) = default;
+
+    /// Return a deep copy of this object.
+    std::shared_ptr<ChebyMap> copy() const { return std::static_pointer_cast<ChebyMap>(_copyPolymorphic()); }
+
+    /**
+    Return the bounding box of the domain of a ChebyMap.
+
+    Return the upper and lower limits of the box defining the domain
+    of either the forward or inverse transformation of a ChebyMap. These
+    are the values that were supplied when the ChebyMap was created.
+
+    If the requested transformation does not have a domain bounding box,
+    this method returns a box determined using MapBox on the opposite transformation,
+    if the opposite transformation is defined.
+
+    @param[in] forward  If true return the domain of the forward transform, else the inverse
+    @throw std::runtime_error if the domain cannot be computed
+    */
+    ChebyDomain getDomain(bool forward) const;
+
+    /**
+    This function creates a new @ref ChebyMap which is a copy of this one,
+    in which a specified transformation (forward or inverse)
+    has been replaced by a new Chebyshev polynomial transformation. The
+    coefficients of the new transformation are estimated by sampling
+    the other transformation and performing a least squares polynomial
+    fit in the opposite direction to the sampled positions and values.
+
+    This method can only be used on (1-input,1-output) or (2-input, 2-output)
+    @ref ChebyMap "ChebyMaps".
+
+    The transformation to create is specified by the `forward` parameter.
+    In what follows "X" refers to the inputs of the @ref ChebyMap, and "Y" to
+    the outputs of the @ref ChebyMap. The forward transformation transforms
+    input values (X) into output values (Y), and the inverse transformation
+    transforms output values (Y) into input values (X). Within a @ref ChebyMap,
+    each transformation is represented by an independent set of
+    polynomials, P_f or P_i: Y=P_f(X) for the forward transformation and
+    X=P_i(Y) for the inverse transformation.
+
+    The `forward` parameter specifies the transformation to be replaced.
+    If it is false, a new forward transformation is created
+    by first finding the input values (X) using the inverse transformation
+    (which must be available) at a regular grid of points (Y) covering a
+    rectangular region of the @ref ChebyMap's output space. The coefficients of
+    the required forward polynomial, Y=P_f(X), are chosen in order to
+    minimise the sum of the squared residuals between the sampled values
+    of Y and P_f(X).
```

Comment:
I'll also note that I intentionally copy the AST docs as close to verbatim as makes sense. But you are right that these details belong separated from the rest, so I moved them. Also, we are free to ask David to improve documentation that is unclear and I have done so in a few cases.